src/components/Image/AsPosts/ImagesAsPostsCard.tsx

```diff
@@ -70,6 +73,10 @@ export function ImagesAsPostsCard({
         modelId: model.id,
         images: [{ id: imageId }],
       }).catch(() => null); // Error is handled in the mutation events
+
+      if (filters.hidden)
+        // Refetch the query to update the hidden images
+        await queryUtils.image.getImagesAsPostsInfinite.invalidate({ ...filters });
```

Comment:
Is there a reason that this is in the `ImagesAsPostsCard` instead of the parent `ImagesAsPostsInfinite`?

Reply:
Basically, since the context menu belongs inside each gallery card, we call the toggleHide mutation here, so it makes sense to invalidate the query whenever we toggle an image